### PR TITLE
[TACHYON-1050] Reenable tachyon.worker.DataServerIntegrationTest 

### PR DESCRIPTION
**Pull request by uronce-cc**
https://tachyon.atlassian.net/browse/TACHYON-1050

### CODE REVIEW

**Comment by uronce-cc** [comment]
In `LocalTachyonCluster` and `LocalTachyonClusterMultiMaster`, `MasterContext` is merged into `WorkerContext`, `WorkerContext` is then merged into `ClientContext`, in integration tests, no matter the user options are set in `MasterContext` or `WorkerContext`, they should finally go into `ClientContext`, but `getClient()` and `getOldClient()` didn't use `ClientContext.getConf()`, instead, it uses `MasterContext.getConf()`, so when user options are set via `WorkerContext`, they are not actually set in clients returned by `getClient()` and `getOldClient()`.

File: integration-tests/src/test/java/tachyon/worker/DataServerIntegrationTest.java [hunk 1]
```diff
@@ -46,6 +46,7 @@
 import tachyon.conf.TachyonConf;
 import tachyon.exception.TachyonException;
 import tachyon.master.LocalTachyonCluster;
+import tachyon.master.MasterContext;
```
Comment (gpang) [inline integration-tests/src/test/java/tachyon/worker/DataServerIntegrationTest.java]: is MasterContext used anywhere?

**Comment by gpang** [comment]
I just had a minor comment. Thanks for fixing this, @uronce-cc !

**Comment by uronce-cc** [comment]
@gpang Fixed, PTAL

**Comment by uronce-cc** [comment]
jenkins, test this please

**Comment by uronce-cc** [comment]
Hmm.... It's still flaky...

**Comment by uronce-cc** [comment]
I misunderstood `TachyonConf.merge`, when `WorkerContext` merges `MasterContext`, existing properties in `WorkerContext` will be overwrote by those in `MasterContext`, so all user or worker properties set via `WorkerContext` in `DataServerIntegrationTest` are overwritten by the default values in `MasterContext`. 

So in integration tests, when properties need to be set before starting the minicluster, they should always be set in `MasterContext`.

File: integration-tests/src/test/java/tachyon/worker/DataServerIntegrationTest.java [hunk 2]
```diff
@@ -36,16 +35,13 @@
 import tachyon.Constants;
 import tachyon.IntegrationTestConstants;
 import tachyon.TachyonURI;
-import tachyon.client.BlockMasterClient;
-import tachyon.client.RemoteBlockReader;
-import tachyon.client.TachyonFSTestUtils;
-import tachyon.client.TachyonStorageType;
-import tachyon.client.UnderStorageType;
+import tachyon.client.*;
```
Comment (gpang) [inline integration-tests/src/test/java/tachyon/worker/DataServerIntegrationTest.java]: I think this should be expanded out?

**Comment by apc999** [comment]
jenkins, test this please

**Comment by apc999** [comment]
PR looks good to me. Just have one more test :)

**Comment by uronce-cc** [comment]
jenkins, test this please

**Comment by uronce-cc** [comment]
jenkins, test this please

**Comment by uronce-cc** [comment]
jenkins, test this please

**Comment by uronce-cc** [comment]
jenkins, test this please

**Comment by gpang** [comment]
LGTM

**Comment by gpang** [comment]
Could you resolve the merge conflicts? Thanks.

**Comment by uronce-cc** [comment]
@gpang conflicts resolved.

**Comment by gpang** [comment]
LGTM

**Pull request merged**